### PR TITLE
Improve Gradle Configuration

### PR DESCRIPTION
**Pull request by Syer10**
- Use Gradle Versions Catalog
- Use type-safe project accessors
- Tachiyomi 1.x was archived so removed the note about it in the gradle files

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+# Logging
+# Stuck on old versions since
+# 1. Logback 1.3.0+ requires Java 9
+# 2. Slf4j 2.0.0+ doesn't register older versions of Logback
+# 3. Kotlin-logging 3.0.2+ requires Java 11, but this is probably a bug
+slf4japi = "org.slf4j:slf4j-api:1.7.32"
+logback = "ch.qos.logback:logback-classic:1.2.6"
+kotlinlogging = "io.github.microutils:kotlin-logging:2.1.21"
+
+#OkHttp
```
Comment (AriaMoradi) [inline gradle/libs.versions.toml]:
```suggestion
# OkHttp
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+config4k = "io.github.config4k:config4k:0.5.0"
+
+# Sort
+sort = "com.github.gpanther:java-nat-sort:natural-comparator-1.1"
+
+# Android stub library
+android-stubs = "com.github.Suwayomi:android-jar:1.0.0"
+
+# Asm modificiation
+asm = "org.ow2.asm:asm:9.4" # must match Dex2Jar version
```
Comment (AriaMoradi) [inline gradle/libs.versions.toml]:
```suggestion
asm = "org.ow2.asm:asm:9.4" # version locked by Dex2Jar
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+rhino-engine = { module = "org.mozilla:rhino-engine", version.ref = "rhino" } # provides the same interface as 'javax.script' a.k.a Nashorn
+
+# Settings
+settings-core = { module = "com.russhwolf:multiplatform-settings-jvm", version.ref = "settings" }
+settings-serialization = { module = "com.russhwolf:multiplatform-settings-serialization-jvm", version.ref = "settings" }
+
+# ICU4J
+icu4j = "com.ibm.icu:icu4j:72.1"
+
+# Image Decoding
```
Comment (AriaMoradi) [inline gradle/libs.versions.toml]:
```suggestion
# Image Decoding implementation provider
```

**Review by AriaMoradi — COMMENTED** [review]
This is going to be hard to read

**Comment by Syer10** [comment]
👍

**Pull request merged**